book.md

```diff
@@ -398,15 +398,76 @@ cache, without the processes having rights to read or write to the same memory
 locations.  To understand how these techniques work, one needs to understand
 typical organization and operation of a cache.
 
-### Typical cache architecture
-
-
-\missingcontent{Add a description of how a cache operates - as far as is
-necessary to explain the cache covert channel techniques.}
-
-### Flush + Reload
-
-### Prime + Probe
+### Typical CPU cache architecture
+
+There is a wide variety in
+[CPU cache micro-architecture](https://en.wikipedia.org/wiki/CPU_cache) details,
+but the main characteristics that are important to set up a covert channel tend
+to be similar across most popular implementations.
+
+Caches are small and much faster memories than the main memory that aim to keep
+a copy of the data at the most frequently accessed addresses in main memory. The
+set of addresses that are used most frequently changes quickly over time as a
+program executes. Therefore, the addresses that are present in CPU caches also
+evolve quickly over time. The content of the cache may change with every
+executed read or write instruction.
+
+On every read and write instruction, the cache micro-architecture looks up if
+the data for the requested address happens to be present in the cache. If it is,
+the CPU can continue executing quickly; if not, dependent operations will have
+to wait until the data returns from the much more slowly main memory.
+Typical access times is 3 to 5 CPU cycles for the fastest cache on a CPU versus
+hundreds of cycles for a main memory access.\index{memory access time}
+
+Most systems have multiple level of cache\index{multi-level cache}, each with a
+different trade-off between cache size\index{cache size} and access
+time\index{cache access time}. Some typical characteristics might be:
+
+* A L1 (level 1) cache, 32kB in size, with an access time of 4 cycles.
+* A L2 cache, 256Kb in size, with an access time of 10 cycles.
+* A L3 cache, 16MB in size, with an access time of 40 cycles.
+* Main memory, gigabytes in size, with an access time of 100 cycles.
+
+When the data was not present in the cache, it typically is stored in the cache
```

Comment:
Thanks for the suggestion - I followed it in the updated patch I'll post soon.
Consistency between "it's" and "it is" is probably something we ought to aim for?

Reply:
I don't have a strong preference myself. I know we are following Chicago style, but I found this APA style blog post an interesting reference point:

https://blog.apastyle.org/apastyle/2015/12/contractions-in-formal-writing-whats-allowed-whats-not.html

My instinct would be to avoid contraction generally but not as a hard rule, as the style is somewhat formal but not completely academic?